test/e2e/framework/ingress_utils.go

```diff
@@ -335,17 +335,26 @@ func createIngressTLSSecret(kubeClient clientset.Interface, ing *extensions.Ingr
 func CleanupGCEIngressController(gceController *GCEIngressController) {
 	pollErr := wait.Poll(5*time.Second, LoadBalancerCleanupTimeout, func() (bool, error) {
 		if err := gceController.Cleanup(false); err != nil {
-			Logf("Still waiting for glbc to cleanup:\n%v", err)
+			Logf("Monitoring glbc's cleanup of gce resources:\n%v", err)
 			return false, nil
 		}
 		return true, nil
 	})
 
+	// Always try to cleanup even if pollErr == nil, because the cleanup
+	// routine also purges old leaked resources based on creation timestamp.
+	By("Performing final delete of any remaining resources")
+	if cleanupErr := gceController.Cleanup(true); cleanupErr != nil {
+		By(fmt.Sprintf("WARNING: possibly leaked resources: %v\n", cleanupErr))
+	} else {
+		By("No resources leaked.")
+	}
+
 	// Static-IP allocated on behalf of the test, never deleted by the
 	// controller. Delete this IP only after the controller has had a chance
 	// to cleanup or it might interfere with the controller, causing it to
 	// throw out confusing events.
-	if ipErr := wait.Poll(5*time.Second, LoadBalancerCleanupTimeout, func() (bool, error) {
+	if ipErr := wait.Poll(5*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
On purpose.  The `LoadBalancerCleanupTimeout` is 15 minutes and is already used above at 336. 

This `wait.Poll(...)` is for manually deleting the static IP created during `Setup(...)`.  We want to try just enough times to make sure any propagation of deletions from above will make the ip viewed as un-used. Even polling for a minute may be overkill.